R/stream.R

```diff
@@ -1,443 +1,230 @@
-#'   timeout = (60 * 10),
-#'   parse = FALSE,
-#'   file_name = "tweets1"
-#' )
-#' stream_tweets(
-#'   timeout = (60 * 10),
-#'   parse = FALSE,
-#'   file_name = "tweets2"
-#' )
-#'
-#' ## parse tweets at a later time using parse_stream function
-#' tw1 <- parse_stream("tweets1.json")
-#' tw1
-#'
-#' tw2 <- parse_stream("tweets2.json")
-#' tw2
-#'
-#' ## streaming tweets by specifying lat/long coordinates
-#'
-#' ## stream continental US tweets for 5 minutes
-#' usa <- stream_tweets(
-#'   c(-125, 26, -65, 49),
-#'   timeout = 300
-#' )
-#'
-#' ## use lookup_coords() for a shortcut verson of the above code
-#' usa <- stream_tweets(
-#'   lookup_coords("usa"),
-#'   timeout = 300
-#' )
-#'
-#' ## stream world tweets for 5 mins, save to JSON file
-#' ## shortcut coords note: lookup_coords("world")
-#' world.old <- stream_tweets(
-#'   c(-180, -90, 180, 90),
-#'   timeout = (60 * 5),
-#'   parse = FALSE,
-#'   file_name = "world-tweets.json"
-#' )
-#'
-#' ## read in JSON file
-#' rtworld <- parse_stream("word-tweets.json")
-#'
-#' ## world data set with with lat lng coords variables
-#' x <- lat_lng(rtworld)
+#' # Download another 10s worth of data to the same file
+#' e <- stream_tweets("election", timeout = 10)
 #'
+#' # stream tweets about continential USA for 5 minutes
+#' usa <- stream_tweets(lookup_coords("usa"), file_name = "usa.json", timeout = 300)
+#' 
 #' }
-#'
-#' @return Tweets data returned as data frame with users data as attribute.
-#' @family stream tweets
+#' @return A tibble with one row per tweet
 #' @export
 stream_tweets <- function(q = "",
                           timeout = 30,
                           parse = TRUE,
                           token = NULL,
                           file_name = NULL,
                           verbose = TRUE,
+                          append = TRUE,
                           ...) {
-  if ("append" %in% names(list(...))) {
-    stop("append should only be used with stream_tweets2() (which is in development)",
-         call. = FALSE)
-  }
-  ## set encoding
-  if (!identical(getOption("encoding"), "UTF-8")) {
-    op <- getOption("encoding")
-    options(encoding = "UTF-8")
-    on.exit(options(encoding = op), add = TRUE)
-  }
-  token <- check_token(token)
-  if (!timeout) {
-    timeout <- Inf
-  }
-  ## setp q to blank for sample stream
-  if (missing(q) || is.null(q)) q <- ""
-  stopifnot(
-    is.numeric(timeout),
-    timeout > 0,
-    any(is.atomic(q), inherits(q, "coords")),
-    is.atomic(file_name)
-  )
-  ## select stream API
-  if (identical(q, "")) {
-    query <- "statuses/sample"
-    params <- NULL
-  } else {
-    query <- "statuses/filter"
-    params <- stream_params(q, ...)
-  }
-  url <- make_url(
-    restapi = FALSE,
-    query,
-    param = params
-  )
-  ## temp file if none provided
   if (is.null(file_name)) {
-    tmp <- TRUE
-    file_name <- tmp_json()
+    file_name <- "stream_tweets.json"
+    inform(paste0("Writing to '", file_name, "'"))
```

Comment:
It could be confusing if people set append = TRUE and see later "Writing to". Perhaps check that the file exists and then adjust the message to "Appending to ...."?
Also the message should be given independently whether the user provides a file name or not.

Reply:
Agreed with appending vs writing, but I think this pattern of only messaging when the user doesn't supply an argument is useful. See https://design.tidyverse.org/def-inform.html for details.